### PR TITLE
Feat/client env file

### PR DESCRIPTION
**Pull request by vikvikvr**
## improve .env file usage

Server was not getting `DB_NAME`  from .env file to create connection string when used in production.

## server: add dev start script

Added script based on `nodemon` for live-realod.

Execute `npm run start:dev` to use it.

## server: add prettier config

Prettier config was missing for the server.

I have set `singleQuote` to false (while on the client is true) to minimize the changes to the existing code.

## weather: improve error handling

Now every status code except for `200` is considered an error, `401` used to break the app.

This emerged by creating a fresh open-weather API key, which requires up to 1 hour to be fully active.

see https://openweathermap.org/faq

### CODE REVIEW

File: server/models/database.js [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-});
+function makeConnectionURI() {
+  const IS_TEST = process.env.NODE_ENV === 'test';
+  const DB_NAME = IS_TEST ? process.env.DATABASE_TEST : process.env.DATABASE;
+
+  if (!DB_USER || !DB_USER_PASSWORD || IS_TEST) {
+    return `mongodb://localhost:27017/${DB_NAME}`;
+  }
+
+  return `mongodb+srv://${DB_USER}:${DB_USER_PASSWORD}@cluster0.9g4e8.mongodb.net/${DB_NAME}?retryWrites=true&w=majority`;
```
Comment (GILREICH1) [inline server/models/database.js:12]: will using a different URI (uses '9g4e8' vs 'kbmrd') not prevent connection the currently used database which holds all the tasks? Maybe this part of the URI should also be dynamic?

**Review by GILREICH1 — CHANGES_REQUESTED** [review]
nice! much cleaner 👍

File: server/models/database.js (hunk 1 above)
Comment (vikvikvr) [reply server/models/database.js:12]: You are absolutely right, I forgot and commited that part as well.

It should come from the .env as well, maybe as `DB_CLUSTER` or something

**Comment by vikvikvr** [comment]
I've added `DB_HOST` to the `.env` file.

Remember to update environment settings on **Heroku** when it's deployed.

Also: moved **Prettier** config to the root folder so it's shared between client and server 😄

File: server/models/database.js (hunk 1 above)
Comment (vikvikvr) [reply server/models/database.js]: I've added `DB_HOST` to the `.env` file.

Remember to update environment settings on **Heroku** when it's deployed.

Also: moved **Prettier** config to the root folder so it's shared between client and server 😄

**Review by GILREICH1 — APPROVED** [review]

**Pull request merged**